### PR TITLE
DDCNL-4983: Added ITs for Agent journey

### PR DESCRIPTION
**Pull request by Christian-QA**

### CODE REVIEW

File: it/controllers/AtsMergePageControllerItSpec.scala [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+
+      result.map(status) mustBe Some(OK)
+
+      request.getQueryString(Globals.TAXS_USER_TYPE_QUERY_PARAMETER) mustBe Some("PORTAL")
+
+      request.getQueryString(Globals.TAXS_AGENT_TOKEN_ID).isDefined mustBe true
+    }
+
+    "return an OK response without query parameters when data is retrieved from backend for both atsList and payeData" in {
+
```
Comment (brendanstraughan) [inline it/controllers/AtsMergePageControllerItSpec.scala:144]: Is this the empty token test? If so can the name of the test say so?

File: it/controllers/AtsMergePageControllerItSpec.scala [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+          get(urlEqualTo(backendUrlSa))
+            .willReturn(aResponse().withStatus(httpResponse))
+        )
+
+        server.stubFor(
+          get(urlEqualTo(backendUrlPaye))
+            .willReturn(aResponse().withStatus(httpResponse))
+        )
+
+        val request = FakeRequest(GET, url)
```
Comment (brendanstraughan) [inline it/controllers/AtsMergePageControllerItSpec.scala:311]: Not sure if this would be overkill but do we need to have tests for the empty token error scenarios as all these error cases are using the url with the token in

File: it/utils/LoginPage.scala [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import org.joda.time.DateTime
+import uk.gov.hmrc.crypto.{AesCrypto, PlainText}
+
+import java.net.URLEncoder
+
+object LoginPage {
+
+  private val encKey = "1111111111111111111111"
+
+  //KxF3antRTEtdaFgpwbmoISnwJDJRvyl0NAnCRwa3SB5EIrpF0IMS/wZwQnvsprKx
```
[the remark this answers is not in the document]
Comment (Christian-QA) [reply it/utils/LoginPage.scala:28]: Done

File: it/controllers/AtsMergePageControllerItSpec.scala (hunk 1 above)
Comment (Christian-QA) [reply it/controllers/AtsMergePageControllerItSpec.scala:144]: Done

File: it/controllers/AtsMergePageControllerItSpec.scala (hunk 2 above)
Comment (Christian-QA) [reply it/controllers/AtsMergePageControllerItSpec.scala:311]: @pascal22p ?

File: it/controllers/AtsMergePageControllerItSpec.scala [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+import play.api.libs.json.Json
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import uk.gov.hmrc.http.HeaderCarrier
+import utils.{FileHelper, Globals, IntegrationSpec, LoginPage}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class AtsMergePageControllerItSpec extends IntegrationSpec with MockitoSugar {
+
```
[the remark this answers is not in the document]
Comment (Christian-QA) [reply it/controllers/AtsMergePageControllerItSpec.scala:38]: Moved a couple vals out, but not sure what else can with file as the tests contained a lot of unique setup that wouldn't be needed in other code, such as the fake application with bound mock, and a unique Auth response with Agent enrolments

File: it/controllers/AtsMergePageControllerItSpec.scala (hunk 2 above)
Comment (pascal22p) [reply it/controllers/AtsMergePageControllerItSpec.scala:311]: No need. This ticket was to unblock an other ticket and this is about the empty token so it will get one.

**Review by brendanstraughan — APPROVED** [review]

**Review by Luka-Abey — APPROVED** [review]
